### PR TITLE
Discord.py 2.0 Migration

### PR DESCRIPTION
**Pull request by MattBSG**
# Pull request template
Use this template for all contributions made for the repository.

Please indicate all that apply with a checkmark. To make a check, convert brackets below to `[x]`.

* [ ] **Feature implementation**
* [x] **Bug fix**
* [x] **Code change/improvement**
* [x] **String change (i.e. grammar/spelling error or an addition)**

***

## Describe what your pull request does
* The bot has been updated to follow new discord.py library functionality and breaking changes.
* We have separated the main modmail bot codebase into it's own, separate, extension and cog so it can actually be quick reloaded like MechaBowser without requiring a full bot power cycle to pull github changes.

## Related issues and extra details
We are no longer user discord-py-slash in any context.

### CODE REVIEW

**Review by mralext20 — APPROVED** [review]

File: cogs/modmail.py [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+    @app_commands.guild_only()
+    @app_commands.default_permissions(view_audit_log=True)
+    async def _reply_anon(self, interaction: discord.Interaction, content: app_commands.Range[str, None, 1800]):
+
+        await self._reply(interaction, content, True)
+
+    async def _reply(self, interaction: discord.Interaction, content, anonymous=False):
+        db = mclient.modmail.logs
+        doc = db.find_one({'channel_id': str(interaction.channel.id)})
+        # Attachments are unable to be sent mod -> user with slash commands (unless it's a url).
```
Comment (7bitlyrus) [inline cogs/modmail.py:100]: Remove

**Review by 7bitlyrus — APPROVED** [review]

File: cogs/modmail.py (hunk 1 above)
Comment (MattBSG) [reply cogs/modmail.py:100]: Working on this, will resolve

**Review by mralext20 — APPROVED** [review]

File: cogs/modmail.py [hunk 2]
```diff
@@ -80,30 +80,43 @@ async def _close_generic(self, user, guild, channel, delay):
...
+        await self._reply(interaction, content, attachment, True)
 
-    async def _reply(self, interaction: discord.Interaction, content, anonymous=False):
+    async def _reply(self, interaction: discord.Interaction, content, attachment, anonymous=False):
         db = mclient.modmail.logs
         doc = db.find_one({'channel_id': str(interaction.channel.id)})
         # Attachments are unable to be sent mod -> user with slash commands (unless it's a url).
         #
         # attachments = [x.url for x in ctx.message.attachments]
         # TODO: If ever able, reenable this functionality
```
Comment (7bitlyrus) [inline cogs/modmail.py]: Remove TODO?

File: cogs/modmail.py [hunk 3]
```diff
@@ -80,30 +80,43 @@ async def _close_generic(self, user, guild, channel, delay):
...
-        attachments = []
-        # if not content and not attachments:
-        #    return await ctx.send('You must provide reply content, attachments, or both to use this command')
+        if not content and not attachment:
+            return await interaction.response.send_message(
+                'You must provide reply content, attachments, or both to use this command', ephemeral=True
```
Comment (7bitlyrus) [inline cogs/modmail.py:114]: Could flow better if written as "To use this command, you must provide either context, an attachment, or both"

**Review by 7bitlyrus — CHANGES_REQUESTED** [review]

**Pull request merged**